ecogvis/functions/misc_dialogs.py

```diff
@@ -1541,9 +1598,1119 @@ def paint(self):
+            vb.setBackgroundColor(
+                (*color, min(elem_alpha, 70)))  # append alpha to color tuple
+            # Main plots
+            mean = p.plot(x=X, y=Y_mean, pen=pg.mkPen((50, 50, 50, min(
+                elem_alpha, 255)), width=1.))
+            semp = p.plot(x=X, y=Y_mean + Y_sem,
+                          pen=pg.mkPen((100, 100, 100, min(elem_alpha, 100)),
+                                       width=.1))
+            semm = p.plot(x=X, y=Y_mean - Y_sem,
+                          pen=pg.mkPen((100, 100, 100, min(elem_alpha, 100)),
+                                       width=.1))
+            fill = pg.FillBetweenItem(semm, semp, pg.mkBrush(100, 100, 100,
+                                                             min(elem_alpha,
+                                                                 100)))
+            p.addItem(fill)
+            p.hideButtons()
+            p.setXRange(X[0], X[-1])
+            yrng = max(abs(Y_mean))
+            p.setYRange(-yrng, yrng)
+            xref = [X[int(len(X) / 2)], X[int(len(X) / 2)]]
+            yref = [-1000 * yrng, 1000 * yrng]
+            p.plot(x=xref, y=yref,
+                   pen=(0, 0, 0, min(elem_alpha, 255)))  # reference mark
+            p.plot(x=X, y=np.zeros(len(X)),
+                   pen=(0, 0, 0, min(elem_alpha, 255)))  # Zero line
+            # Axis control
+            left = p.getAxis('left')
+            left.setStyle(showValues=False)
+            left.setTicks([])
+            bottom = p.getAxis('bottom')
+            bottom.setStyle(showValues=False)
+            bottom.setTicks([])
+        # store scale limits
+        self.Yscale['global max'] = [ymin, ymax]
+        self.Yscale['global std'] = [-ystd, ystd]
+
+    def areas_select(self):
+        # Dialog to choose channels from specific brain regions
+        w = SelectChannelsDialog(self.parent.model.all_regions,
+                                 self.parent.model.regions_mask)
+        self.transparent = []
+        for ind, ch in enumerate(self.grid_order):
+            loc = self.parent.model.nwb.electrodes['location'][ch]
+            if loc not in w.choices:
+                self.transparent.append(ch)
+        self.draw_erp()
+
+
+# Gray line for visual separation of buttons --------------------------------
 class QHLine(QtGui.QFrame):
     def __init__(self):
         super().__init__()
         self.setFrameShape(QtGui.QFrame.HLine)
         self.setFrameShadow(QtGui.QFrame.Sunken)
+
+
+# Viewbox for ERP plots -----------------------------------------------------
+class CustomViewBox(pg.ViewBox):
+    def __init__(self, parent, ch):
+        pg.ViewBox.__init__(self)
+        self.parent = parent
+        self.ch = ch
+
+    def mouseDoubleClickEvent(self, ev):
+        IndividualERPDialog(self)
+
+
+# Individual Event-Related Potential dialog ---------------------------------
+class IndividualERPDialog(QtGui.QDialog):
```

Comment:
this shouldn't be here, it has been moved out to its own module, 'event_related_potential.py'